app/controllers/tokens_controller.rb

```diff
@@ -4,10 +4,11 @@ class TokensController < ApplicationController
   before_action :ensure_token_auth_enabled!
 
   def create
-    @token = Token.new(token_params)
-    if @token.save
-      TokenMailer.new_token_email(@token).deliver_later
-      render
+    @token = generate_token(token_params)
+    if @token.nil?
+      show_throttle_limit_error
```

Comment:
This still shows the throttle limit error if the email is invalid (i.e. if `Token.create` fails), doesn't it? I think using `nil` to mean throttled is inherently problematic - at least if we want to distinguish error cases on the front end.